teachopencadd/talktorials/T032_compound_activity_proteochemometrics/talktorial.ipynb

```diff
@@ -0,0 +1,1920 @@
+{
```

Comment:
<p>Moldred &gt; Mordred (maybe do a Ctrl + F to get and rule them all)</p><p>also in</p><p><code>moldred_descriptors</code> and <code>moldred_list</code></p><p>--</p><p>Again, <code>bioactivity_dataset.SMILES</code> to <code>bioactivity_dataset["SMILES"]</code></p><p>--</p><p>Why 2,147,483,647&nbsp;maximum (could be added to the comment)</p><p>--</p><p>I suggest to do only one cleaning step per row (sometimes there are 2) while adding a comment before each code line.</p><p>Do not use <code>inplace=True</code> here to make the code more uniform.</p><p>Here I tried to split the code into the respective actions - could you please correct where I got lost?</p><pre class="ql-syntax" spellcheck="false"># Format descriptor data

# - Renaming duplicated columns (can you give an example of a duplicate?
# QUESTION: Why do we call .descriptors? Should that not be simply the column names (as in .columns)?
# QUESTION: Can we please do this in two steps instead of the nested re.sub usage?
mordred_descs_names = {
&nbsp; &nbsp; str(x): re.sub(r'(.*F?)A(H?Ring)$', r'\1aliph\2', re.sub(r'(.*F?)a(H?Ring)$', r'\1arom\2', str(x))) 
    for x 
    in Calculator(descriptors, ignore_3D=True).descriptors
}
molecular_descriptor = pd.DataFrame(molecular_descriptor.rename(mordred_descs_names))

# - Replacing values bigger than 2,147,483,647 by 0
# QUESTION: Where do we do this? here? I am trying to split code into single operations per cell
molecular_descriptor = molecular_descriptor.fill_missing(np.NAN) # Why this step? Missing values are not np.NAN here?
molecular_descriptor = molecular_descriptor.astype(np.float32)
molecular_descriptor = molecular_descriptor.replace([np.inf, -np.inf], np.NAN)
molecular_descriptor = molecular_descriptor.fillna(value=0)

# - Rounding values to 3 decimals
molecular_descriptor = molecular_descriptor.round(3)

# - Converting to minimal memory footprint
molecular_descriptor = molecular_descriptor.convert_dtypes()

# - Inserting SMILES in first column
molecular_descriptor.insert(0, 'SMILES', bioactivity_dataset.SMILES.unique())
</pre><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/278/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='63'/>

Reply:
<p>Updated it to make it more comprehensive, I hope now it's better :) </p>